contrib/lock/client/client.go

```diff
@@ -118,90 +83,63 @@ func write(key string, value string, version int64) error {
 	return nil
 }
 
-func read(key string) (string, int64) {
-	req := request{
-		Op:  "read",
-		Key: key,
-	}
-
-	reqBytes, err := json.Marshal(&req)
-	if err != nil {
-		fmt.Printf("failed to marshal request: %s\n", err)
-		os.Exit(1)
-	}
-
-	httpResp, err := http.Post("http://localhost:8080", "application/json", bytes.NewReader(reqBytes))
-	if err != nil {
-		fmt.Printf("failed to send a request to storage: %s\n", err)
-		os.Exit(1)
-	}
-
-	respBytes, err := io.ReadAll(httpResp.Body)
-	if err != nil {
-		fmt.Printf("failed to read request body: %s\n", err)
-		os.Exit(1)
-	}
-
-	resp := new(response)
-	err = json.Unmarshal(respBytes, resp)
-	if err != nil {
-		fmt.Printf("failed to unmarshal response json: %s\n", err)
-		os.Exit(1)
-	}
-
-	return resp.Val, resp.Version
-}
-
 func main() {
 	if len(os.Args) != 2 {
-		fmt.Printf("usage: %s <1 or 2>\n", os.Args[0])
-		return
+		log.Fatalf("usage: %s <1 or 2>", os.Args[0])
 	}
 
 	mode, err := strconv.Atoi(os.Args[1])
 	if err != nil || mode != 1 && mode != 2 {
-		fmt.Printf("mode should be 1 or 2 (given value is %s)\n", os.Args[1])
-		return
+		log.Fatalf("mode should be 1 or 2 (given value is %s)", os.Args[1])
 	}
 
-	fmt.Printf("client %d starts\n", mode)
+	log.Printf("client %d starts\n", mode)
```

Comment:
minor comment: the `\n` isn't needed because the lib will automatically append a `\n ` if it doesn't exist.  Of course it isn't a big deal.

Reply:
Note all `log.XXX` functions do not need the `\n`. Again, it isn't a big deal, feel free to resolve it in a separate PR if you feel comfortable.